SimPEG/maps.py

```diff
@@ -1892,6 +1895,10 @@ class ParametricSplineMap(IdentityMap):
     slope = 1e4
 
     def __init__(self, mesh, pts, ptsv=None, order=3, logSigma=True, normal="X"):
+        if isinstance(mesh, discretize.CurvilinearMesh):
```

Comment:
good to put this check. curious why you are so into this? May be because you are actively using CurvilinearMesh?

Reply:
Hah, yes, good guess. I feel it is criminally underutilized, for my application it is such an elegant solution.